packages/plugins/newrelic/src/index.ts

```diff
@@ -125,50 +119,75 @@ export const useNewRelic = (rawOptions?: UseNewRelicOptions): Plugin => {
         );
       }
     },
-    async onExecute({ args }) {
-      const instrumentationApi = await instrumentationApi$;
-      const transactionNameState = instrumentationApi.agent.tracer.getTransaction().nameState;
-      const spanContext = instrumentationApi.agent.tracer.getSpanContext();
-      const delimiter = transactionNameState.delimiter;
-      const rootOperation = args.document.definitions.find(
-        // @ts-expect-error TODO: not sure how we will make it dev friendly
-        definitionNode => definitionNode.kind === Kind.OPERATION_DEFINITION
-      ) as OperationDefinitionNode;
+    onParse({ params: { source } }) {
+      return ({ result }) => {
+        if (options.includeOperationDocument) {
+          const key = source instanceof Source ? source.body : source;
+
+          if (!result || result instanceof Error) return;
+
+          documentSourceMap.set(result, key);
+        }
+      };
+    },
+    onExecute({ args }) {
+      const rootOperation = getOperationAST(args.document, args.operationName);
+      if (!rootOperation) {
+        logger.trace('No root operation found. Not recording transaction.');
+        return;
+      }
       const operationType = rootOperation.operation;
-      const document = print(args.document);
       const operationName =
         options.extractOperationName?.(args.contextValue) ||
         args.operationName ||
         rootOperation.name?.value ||
         AttributeName.ANONYMOUS_OPERATION;
-      let rootFields: string[] | null = null;
 
-      if (options.rootFieldsNaming) {
-        const fieldNodes = rootOperation.selectionSet.selections.filter(
-          selectionNode => selectionNode.kind === Kind.FIELD
-        ) as FieldNode[];
-        rootFields = fieldNodes.map(fieldNode => fieldNode.name.value);
+      const transaction = instrumentationApi.agent.tracer.getTransaction();
+      const transactionNameState = transaction?.nameState;
+      if (transactionNameState) {
+        const delimiter = transactionNameState.delimiter || '/';
+        let rootFields: string[] | null = null;
+
+        if (options.rootFieldsNaming) {
+          const fieldNodes = rootOperation.selectionSet.selections.filter(
+            selectionNode => selectionNode.kind === Kind.FIELD
+          ) as FieldNode[];
+          rootFields = fieldNodes.map(fieldNode => fieldNode.name.value);
+        }
+
+        const operationType = rootOperation.operation;
+        const operationName =
+          options.extractOperationName?.(args.contextValue) ||
+          args.operationName ||
+          rootOperation.name?.value ||
+          AttributeName.ANONYMOUS_OPERATION;
```

Comment:
I think this variable has been defined at the top with the same definition and is accessible from line 140
```suggestion
```